x-pack/legacy/plugins/siem/public/components/embeddables/embedded_map.tsx

```diff
@@ -158,20 +158,32 @@ export const EmbeddedMap = React.memo<EmbeddedMapProps>(
       return () => {
         isSubscribed = false;
       };
-    }, [loadingKibanaIndexPatterns, kibanaIndexPatterns]);
+    }, [
+      loadingKibanaIndexPatterns,
+      kibanaIndexPatterns,
+      dispatchToaster,
+      endDate,
+      filters,
+      plugins.embeddable,
+      portalNode,
+      query,
+      setQuery,
+      siemDefaultIndices,
+      startDate,
+    ]);
```

Comment:
The `embeddableObject` is an expensive component to create, and only needs to be created once per loading of `kibanaIndexPatterns`. All these other dependencies are really only necessary for initialization. If they later change (e.g. `filters` or `startDate`), there are specific `useEffect`'s below (with corresponding dependencies) for dispatching those state changes. Will need to `ts-ignore` the warning for `react-hooks/exhaustive-deps`.